src/applications/virtual-agent/components/webchat/helpers/startConvoAndTrackUtterancesHelpers.js

```diff
@@ -0,0 +1,104 @@
+import * as _ from 'lodash';
+import {
+  joinActivity,
+  startConversationActivity,
+} from '../../../actions/index';
+import piiReplace from '../piiReplace';
+import {
+  IN_AUTH_EXP,
+  IS_TRACKING_UTTERANCES,
+  LOGGED_IN_FLOW,
+  RECENT_UTTERANCES,
+} from '../../chatbox/utils';
+
+// define thunks for actions
+export const processActionConnectFulfilled = ({
+  requireAuth,
+  dispatch,
+  csrfToken,
+  apiSession,
+  apiURL,
+  baseURL,
+  userFirstName,
+  userUuid,
+}) => () => {
+  if (requireAuth && sessionStorage.getItem(LOGGED_IN_FLOW) !== 'true') {
+    dispatch(
+      startConversationActivity(
+        csrfToken,
+        apiSession,
+        apiURL,
+        baseURL,
+        userFirstName,
+        userUuid,
+      ),
+    );
+  }
+  dispatch(joinActivity);
```

Comment:
Reaffirming here that the join activity should occur under `processActionConnectFulfilled` regardless of conditionals. This is in the correct place.